### ISSUE TITLE
Support images with a repository/container name component

### ISSUE THREAD

**Issue by fsmunoz** [gramineproject/gsc#51]
## Description of the problem

Some images can have a registry and repository component to their names. It would be good if GSC supported them when building images.

As an example, the Red Hat UBI image for Python 3.8 is at `registry.access.redhat.com/ubi8/python-38`.

## Steps to reproduce

Add something like this to `config.yaml`

`Distro: "registry.access.redhat.com/ubi8/python-38`

An additional variable could also be used to store the new path component (see _Additional information_). Note that this example uses a non-supported image (that will be a separate issue/PR), but the idea is the same for e.g. `my.private.registry.cloud/dev/centos`, which would use the `centos` templates but include the full path in the `FROM`.

## Expected results

The build process should understand that structure and use in e.g. the `FROM` fields of the `Dockerfile`, and the build process should continue normally.

## Actual results

There is currently no support for this so it fails

```
$ ./gsc build  registry.access.redhat.com/ubi8/python-38 test/generic.manifest 
Building unsigned graminized Docker image `gsc-registry.access.redhat.com/ubi8/python-38-unsigned` from original application image `registry.access.redhat.com/ubi8/python-38`...
Traceback (most recent call last):
  File "./gsc", line 12, in <module>
    sys.exit(main(sys.argv))
  File "/home/frmuno/orig/og/gsc/./gsc.py", line 476, in main
    return args.command(args)
  File "/home/frmuno/orig/og/gsc/./gsc.py", line 179, in gsc_build
    distro, _ = distro.split(':')
ValueError: not enough values to unpack (expected 2, got 1)
```

**Comment by fsmunoz** [comment]
Added https://github.com/gramineproject/gsc/pull/52 to address this; it addas a `Registry`variable which is commented out in the template and, if used, is prependend in the `FROM` in the `Dockerfile`.

**Comment by dimakuv** [comment]
Thanks @fsmunoz! I left my comments in that PR #52.

**Comment by dimakuv** [comment]
@fsmunoz Please specify in the PR description that `Fixes #51`. This will make the PR to automatically close this issue once the PR is merged.

**Comment by fsmunoz** [comment]
@dimakuv Thank you, will do, forgot the "fixes" somehow :)

**Comment by mkow** [comment]
@fsmunoz: To the description, not the title ;) See https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword.
